### PR TITLE
Fixed an issue preventing chat message to be reported in some versions of Skype for Business.

### LINKED ISSUE #9295: nvda does not automatically read messages received on skype for business
### Steps to reproduce:
I'm sure I've seen this working for some time.
I have not used skype for business for a long time but in my current work I'm making use of it again, and when I received a message on skype for business, nvda did not automatically read the received message.
Open a conversation on skype for business type something and wait inside the conversation for your partner to respond.
### Actual behavior:
The nvda is mute.
### Expected behavior:
Nvda should automatically announce the response received.
#### NVDA installed/portable/running from source:
install
#### NVDA version:
2018.4.1
#### Windows version:
10 17134.556
#### Name and version of other software in use when reproducing the issue:
office 16.0.4266.1001

#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA?
no

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:

Fixes #9295

### Summary of the issue:

In Skype for Business (at least Office 2016, version 16.0.5095.1000), the incoming messages are not reported and an error occurs in the log instead.
This error show that the message has a value set to None whereas the code expects a string in order to parse it.
(cf. https://github.com/nvaccess/nvda/issues/9295#issuecomment-785114245 for more details)

### Description of how this pull request fixes the issue:

* If the message value is None, parse only the message name to split the content from author and timestamp parts. For this, we just look at ", , " separators.
* If the message is not None, use as previously the message name and the message value to split author, content and timestamp of the message.

I suppose that the message value may be None or not according to Skype / Lync version. I do not had the opportunity to check however on other Skype / Lync version.

I did not remove the old way to do because of the following comment:
> `# Example string: "Michael Curran : , , Hello\r\n\r\nThis is a test , 10:45 am."`
In this comment, we can see that there is no ', , ' separator between content and timestamp. Did it change between various versions of Skype / Lync? Or is the example in the comment just not correct? Since I do not know, I have kept the two paths.

### Testing strategy:

* Tested on Skype for Business, Office 2016, version 16.0.5095.1000).
* Would need to be tested on other versions: Lync (Office 2013), Office 365
  However, since I have not found someone to test on other versions of Skype for Business, the old path (when message value is not None) will be tested on alpha. I have taken care however to keep as much as possible the old path code in order to limit risks of regression.
* It does not seem to me that unit tests are required for such modifications.
* I do not think that we have an environment to perform system tests on Skype for Business or more generally Office.

### Known issues with pull request:

Missing tests on other versions of Office due to lakc of testers; will be tested on alpha.

### Change log entry:

Section: Bug fixes
`The incoming messages in the chat of Skype for Business are reported again. (#9295)`

### Code Review Checklist:

This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review and confirm you have considered the following items.
Mark items you have considered by checking them.
You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
You can also check the checkboxes after the PR is created.

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
Looking for other people that may test this PR in other versions of Office:
* Office 2013 / Lync
* Office 2016 more recent than mine
* Office 365
Thanks!
Also cc @fernando-jose-silva:

Could you test on your side that the build in this PR resolves #9295?

**Comment by CyrilleB79** [comment]
@Qchristensen would you be able to test it?
Or could you Cc people that may be able to do this test on various versions of Office?
Thanks.

**Comment by CyrilleB79** [comment]
Also a reminder for @fernando-jose-silva: if you can test [this build](https://ci.appveyor.com/api/buildjobs/qv6b2h08do65mbtt/artifacts/output%2Fnvda_snapshot_pr12122-21816%2C35ac424f.exe) and indicate your version of Office/Skype. Thanks.

**Comment by fernando-jose-silva** [comment]
I'm sorry, unfortunately I no longer use Skype for business where I work, the organization has already moved to teams, I have no way to test it.
Thanks.

Enviado do Email para Windows 10

De: Cyrille Bougot
Enviado:quarta-feira, 10 de março de 2021 17:49
Para: nvaccess/nvda
Cc:fernando-jose-silva; Mention
Assunto: Re: [nvaccess/nvda] Fixed an issue preventing chat message to bereported in some versions of Skype for Business. (#12122)

Also a reminder for @fernando-jose-silva: if you can test this build and indicate your version of Office/Skype. Thanks.

**Comment by CyrilleB79** [comment]
@michaelDCurran: since you are the original author of the Lync app module (in #7287) could you please do the tests made when introducing #7287 in order to check that this PR does not break anything.

Marking this PR as ready since everything is made on my side.
If some additional tests are required, just let me know if you know someone else that could do it.
Thanks.

File: source/appModules/lync.py [hunk 1]
```diff
@@ -18,15 +18,23 @@ def event_liveRegionChange(self):
...
-		# Normalize the usage of end-of-line characters (name and value seem to expose them differently, which would break comparison)
-		content=self.value.replace('\r','\n').strip()
-		fullText=self.name.replace('\r\n\r\n','\n')
+		fullText = self.name.replace('\r\n\r\n', '\n')
+		if self.value is None:
+			# For some versions of Skype for Business, self.value is just None.
+			# So we just look at self.name formatting to split content from person and timestamp.
+			contentStartIndex = self.name.find(', , ') + 4
+			contentEndIndex = self.name.rfind(', , ')
+			content = self.name[contentStartIndex:contentEndIndex].replace('\r\n\r\n', '\n').strip()
```
Comment (seanbudd) [inline source/appModules/lync.py:27]: could this be replaced with a regex match, and an logged error if this can't be found properly?

**Review by seanbudd — CHANGES_REQUESTED** [review]
Thanks for this and sorry for the delay in reviewing. I think this is a relatively safe change as content where `self.value is None` is going to be broken right now regardless and this will only improve handling. It would be good to log some information that allows us to figure out how to extract `content` if a version of Lync doesn't match that formatting.

**Comment by lukaszgo1** [comment]
@CyrilleB79  Is this still on your radar?

**Comment by CyrilleB79** [comment]
@lukaszgo1 thanks for the reminder. I must confess that I had not imagined that this issue would have impacted other people that may have updated their version of Office. Thus I had let it apart...

I will perform the corrections and do the tests when I have the opportunity.

**Comment by CyrilleB79** [comment]
@seanbudd your comments should be addressed now. Sorry for the delay.

**Review by seanbudd — APPROVED** [review]
Thanks @lukaszgo1

**Pull request merged**